From 51e20efbf13214c167fb01f8a5920eb344b00115 Mon Sep 17 00:00:00 2001
From: Emergi Idan <idan.emergi@intel.com>
Date: Thu, 5 Dec 2019 15:32:17 +0200
Subject: [PATCH] PONRTSYS-6252: Fix PPv4 LRO info in SI

Fix session manager handling of the template UD and
protocol specific

Signed-off-by: Emergi Idan <idan.emergi@intel.com>
---
 drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c b/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
index 620a4386a706..e2e9f4b057bf 100644
--- a/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
+++ b/drivers/net/ppv4/pp-api/pp-session-mgr/pp_session_mgr.c
@@ -2676,14 +2676,13 @@ static s32 __smgr_si_ud_set(struct sess_info *sess)
 		sess->si.si_ud_sz = sess->args->ps_sz + sess->args->ud_sz;
 
 	/* if this is a session for fragments, make sure UD will be copied
-	 * for the egress UC
+	 * for the egress UC, otherwise, use the user arguments
 	 */
-	if (SESS_INGRS_PKT(sess)) {
-		if (SESS_INGRS_OUTER_BMAP(sess) & FV_L3_FRAG_1ST ||
-		    SESS_INGRS_INNER_BMAP(sess) & FV_L3_FRAG_1ST) {
-			sess->si.tmpl_ud_sz = PP_TEMPLATE_UD_SZ;
-			sess->si.ps_copy = true;
-		}
+	if (SESS_INGRS_PKT(sess) &&
+	    (SESS_INGRS_OUTER_BMAP(sess) & FV_L3_FRAG_1ST ||
+	     SESS_INGRS_INNER_BMAP(sess) & FV_L3_FRAG_1ST)) {
+		sess->si.tmpl_ud_sz = PP_TEMPLATE_UD_SZ;
+		sess->si.ps_copy = true;
 	} else {
 		sess->si.tmpl_ud_sz = sess->args->tmp_ud_sz;
 		sess->si.ps_copy =
